dev/tests/api-functional/testsuite/Magento/GraphQl/CorsHeadersTest.php

```diff
@@ -0,0 +1,126 @@
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_MAX_AGE, '86400');
+        $this->reinitConfig->reinit();
+
+        $headers = $this->getHeadersFromIntrospectionQuery();
+
+        self::assertArrayNotHasKey('Access-Control-Max-Age', $headers);
+        self::assertArrayNotHasKey('Access-Control-Allow-Credentials', $headers);
+        self::assertArrayNotHasKey('Access-Control-Allow-Headers', $headers);
+        self::assertArrayNotHasKey('Access-Control-Allow-Methods', $headers);
+        self::assertArrayNotHasKey('Access-Control-Allow-Origin', $headers);
+    }
+
+    public function testCorsHeadersWhenCorsIsEnabled()
+    {
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_HEADERS_ENABLED, 1);
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_ALLOWED_HEADERS, 'Origin');
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_ALLOW_CREDENTIALS, '1');
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_ALLOWED_METHODS, 'GET,POST');
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_ALLOWED_ORIGINS, 'magento.local');
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_MAX_AGE, '86400');
+        $this->reinitConfig->reinit();
+
+        $headers = $this->getHeadersFromIntrospectionQuery();
+
+        self::assertEquals('Origin', $headers['Access-Control-Allow-Headers']);
+        self::assertEquals('1', $headers['Access-Control-Allow-Credentials']);
+        self::assertEquals('GET,POST', $headers['Access-Control-Allow-Methods']);
+        self::assertEquals('magento.local', $headers['Access-Control-Allow-Origin']);
+        self::assertEquals('86400', $headers['Access-Control-Max-Age']);
+    }
+
+    public function testEmptyCorsHeadersWhenCorsIsEnabled()
+    {
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_HEADERS_ENABLED, 1);
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_ALLOWED_HEADERS, '');
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_ALLOW_CREDENTIALS, '');
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_ALLOWED_METHODS, '');
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_ALLOWED_ORIGINS, '');
+        $this->resourceConfig->saveConfig(Configuration::XML_PATH_CORS_MAX_AGE, '');
+        $this->reinitConfig->reinit();
+
+        $headers = $this->getHeadersFromIntrospectionQuery();
+
+        self::assertArrayNotHasKey('Access-Control-Max-Age', $headers);
+        self::assertArrayNotHasKey('Access-Control-Allow-Credentials', $headers);
+        self::assertArrayNotHasKey('Access-Control-Allow-Headers', $headers);
+        self::assertArrayNotHasKey('Access-Control-Allow-Methods', $headers);
+        self::assertArrayNotHasKey('Access-Control-Allow-Origin', $headers);
+    }
+
+    private function getHeadersFromIntrospectionQuery()
```

Comment:
The method return type is missed.

Reply:
we don't usually do it on tests, but since we upgraded phpunit we might have to

Reply:
It doesn't matter if a method is a test class method or a regular code. All code should follow the same semantic, the only exception is doc blocks, they can be omitted for test methods.